### DISCUSSION TITLE
About saving the osg::MatrixTransform parameter.

### DISCUSSION THREAD

**Discussion by luohuiiqng** [openscenegraph/OpenSceneGraph#1314] [Q&A]
Hello! 
Now I'm trying to save the parameters of a matrix for the next run of the program, but the rotation error in using the saved data is very large and it is skewed in a fixed direction every time. What am I doing wrong, I am getting the saved data and using the block of code that saves the data. Thank you!

```
NodeAxle::PorpertyData NodeAxle::saveProperty()
{
    osg::Vec3d position = m_transform->getMatrix().getTrans();
    osg::Vec3d scale = m_transform->getMatrix().getScale();
    osg::Quat quat = m_transform->getMatrix().getRotate();

    osg::Vec4  q4(quat._v[0], quat._v[1], quat._v[2], quat._v[3]);

    PorpertyData data;
    data._position = position;
    data._scale = scale;
    data._q4 = q4;
    data._path = m_path.c_str();
    data._uuid = m_uuid.c_str();
    return data;
}


void NodeAxle::setProperty(PorpertyData data)
{
    osg::Matrix  S = osg::Matrix::scale(data._scale);
    osg::Matrix  R = osg::Matrix::rotate(osg::Quat(data._q4));
    osg::Matrix  T = osg::Matrix::translate(data._position);   
    m_transform->setMatrix(R * S * T);
}
```
Adjusted posture:
![ed6cace8a26b78a1bf54307e3f4791e](https://github.com/openscenegraph/OpenSceneGraph/assets/82392471/95244897-76c7-4fde-918d-ac2de04d43e4)

The gesture after calling setProperty:

![10e4e59a4b30e49944455d06a6f521e](https://github.com/openscenegraph/OpenSceneGraph/assets/82392471/148f074a-c34f-4937-932a-7b2dd674fb65)

**Comment by luohuiiqng** [comment]
Through observation I find that the problem is related to scaling, if not scaling then using the saved parameter node attitude is normal, why? Thank you!

**Comment by xarray** [comment]
Hi, please consider use Matrix::decompose() instead of just getRotate() / getScale().
